### PR TITLE
Fix handling carriage return on linux

### PR DESCRIPTION
**Pull request by mingodad**

### CODE REVIEW

**Comment by ricomariani** [comment]
I knew that CRLF (\r\n) was a thing on DOS but LFCR (\n\r) on Linux?  Every *nix I've ever used uses \n only.

What's going on here?

File: sources/cql.l [hunk 1]
```diff
@@ -218,13 +218,14 @@ static bool_t cql_already_processed_file(CSTR str) {
 
 stop [^A-Z_0-9]
 sp [ \t]+
+nl \r\n?|\n\r?
```
Comment (ricomariani) [inline sources/cql.l:221]: This matches \r or \r\n or \n\r or \n  Two of which are not standard line endings and one is only used on DOS?  What's going on here?

**Comment by mingodad** [comment]
When on linux we try to parse/use a text file that has `\r\n` `cql` get's confused.

File: sources/cql.l (hunk 1 above)
Comment (ricomariani) [reply sources/cql.l:221]: I think \n\r is just wrong.

**Comment by ricomariani** [comment]
I can understand that if you try to read a DOS/Windows file on linux it will get confused but that is only \r\n

There is no world where \n\r is legal.  That has to go.   The valid choices are LF or CRLF, only.

R
________________________________

File: sources/cql.l (hunk 1 above)
Comment (ricomariani) [reply sources/cql.l:221]:
```
nl \r\n|\n
```

works for me

File: sources/cql.l [hunk 2]
```diff
@@ -635,7 +636,7 @@ X'({hex}{hex})*'                 { yylval.sval = Strdup(yytext); return BLOBLIT;
                                    return result;
                                  }
 
-[ \t\n]                          ;
+[ \t\n\r]                        ;
```
Comment (ricomariani) [inline sources/cql.l:639]: This is for sure needed

**Comment by mingodad** [comment]
https://en.wikipedia.org/wiki/Newline 
```
[Acorn BBC](https://en.wikipedia.org/wiki/BBC_Micro)[[6]](https://en.wikipedia.org/wiki/Newline#cite_note-6) and [RISC OS](https://en.wikipedia.org/wiki/RISC_OS) spooled text output[[7]](https://en.wikipedia.org/wiki/Newline#cite_note-7)	LF CR	0A 0D	10 13	\n\r
```

File: sources/cql.l [hunk 3]
```diff
@@ -306,15 +307,15 @@ reallit ({d}+"."{d}*|"."{d}+)(E("+"|"-")?{d}+)?
...
 <skipping_c_comment>"*/"         { BEGIN(skipping); }
 
-<skipping>--.*\n
+<skipping>--.*{nl}
 <skipping>{qid}
 <skipping>{strlit}
 <skipping>{cstrlit}
 <skipping>.
-<skipping>[\n]
+<skipping>{nl}
```
Comment (ricomariani) [inline sources/cql.l:318]: This is not needed because the \r comes before the \n always. if it is present.

File: sources/cql.l [hunk 4]
```diff
@@ -306,15 +307,15 @@ reallit ({d}+"."{d}*|"."{d}+)(E("+"|"-")?{d}+)?
 
 <skipping>"/*"                   { BEGIN(skipping_c_comment); }
 <skipping_c_comment>.
-<skipping_c_comment>[\n]
+<skipping_c_comment>{nl}
 <skipping_c_comment>"*/"         { BEGIN(skipping); }
 
-<skipping>--.*\n
+<skipping>--.*{nl}
```
Comment (ricomariani) [inline sources/cql.l:313]: This is not needed because the \r comes before the \n always. if it is present.

File: sources/cql.l [hunk 5]
```diff
@@ -306,15 +307,15 @@ reallit ({d}+"."{d}*|"."{d}+)(E("+"|"-")?{d}+)?
 
 <skipping>"/*"                   { BEGIN(skipping_c_comment); }
 <skipping_c_comment>.
-<skipping_c_comment>[\n]
+<skipping_c_comment>{nl}
```
Comment (ricomariani) [inline sources/cql.l:310]: This is not needed because the \r comes before the \n always. if it is present.

File: sources/cql.l [hunk 6]
```diff
@@ -218,13 +218,14 @@ static bool_t cql_already_processed_file(CSTR str) {
...
 sp [ \t]+
+nl \r\n?|\n\r?
 hex [0-9A-F]
 d [0-9]
 
 id [_A-Z][A-Z0-9_]*
-qid `(``|[^`\n])*`
-cstrlit \"(\\.|[^\\"\n])*\"
-strlit '(''|[^']|\n)*'
+qid `(``|[^`\n\r])*`
```
Comment (ricomariani) [inline sources/cql.l:226]: These are needed

**Comment by ricomariani** [comment]
Yeah not interested in BBC Micro.  It was non standard even then and it's a 30 year old system?

________________________________

**Comment by ricomariani** [comment]
Only one line of this was needed and I added test cases https://github.com/ricomariani/CG-SQL-author/commit/d405eeb1df704d8fb46034f6973462533c88cd53

**Pull request closed without merging**